src/util/file_reader.js

```diff
@@ -118,13 +120,15 @@ class NewlineReader {
                         console.warn('line too long finally terminated at eof:', this.info());
                     } else {
                         const line = this.buf.toString('utf8', this.start, this.end);
+                        this.start = this.end;
```

Comment:
I know the intention here to set `this.start` was to make sure that we can keep reading while the file was being written but I think to enable that we would need more than just this as you would notice that we enter the read loops only if `this.eof === false`. Not something to be addressed in this PR probably, or maybe just a comment on top of the loop stating that after each read the reader must set `reader.eof = false` if they intend to keep on reading from a file while it is being written.